### PR TITLE
Add support for building with CMake

### PR DESCRIPTION
**Pull request by twam**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,92 @@
+cmake_minimum_required(VERSION 3.15)
+
+set(AVR_MCU attiny44 CACHE STRING "MCU")
+set(AVR_L_FUSE 0xE2 CACHE STRING "Low Fuse")
+set(AVR_H_FUSE 0xDD CACHE STRING "High Fuse")
+set(AVR_E_FUSE 0xFE CACHE STRING "Extended Fuse")
+set(AVRDUDE_PROGRAMMER avrispmkII CACHE STRING "Programmer")
```
Comment (hayhut) [inline CMakeLists.txt:7]: Edit readme to describe what to change in order to target a different AVR programmer.

File: README.md [hunk 2]
```diff
@@ -106,13 +106,21 @@ configure a different parameter. For each enable "i" the bitfield contains:
...
 This firmware is written for use with AVR GCC 5.4 and AtmelStudio 7. To compile
 the firmware in Atmel Studio, create a new project and add the .c files in the
 `src` directory to the project. The project Toolchain settings must be modified
 to include the `include` directory containing the firmware header files. With
 the project configured correctly it should be possible to build and debug using
 tools provided by Atmel Studio.
 
+### CMake
+
+Alternatively the firmware can be built using [CMake](https://cmake.org)
```
Comment (hayhut) [inline README.md:120]: Edit readme to describe a general process for what a user would do with the CMake file and how to use it (commands, etc).

**Review by hayhut — CHANGES_REQUESTED** [review]
Hello and thank you for contributing to SYZYGYfpga! I just had a few suggestions. Otherwise, this is great stuff and we appreciate all of your recent contributions.

**Comment by twam** [comment]
Hey @hayhut,

I extended the documentation in the README. Let me know if it's still unclear.

File: README.md [hunk 3]
```diff
@@ -106,13 +106,39 @@ configure a different parameter. For each enable "i" the bitfield contains:
...
+and any AVR GCC version. If [`avrdude`](https://www.nongnu.org/avrdude/)
+is detected by CMake, targets for flashing (`flash`), writing
+EEPROM (`eeprom`) and setting fuses (`fuse`) are also available.
+
+Use the following commands to built the firmware with CMake:
+```
+https://github.com/SYZYGYfpga/avr-dna-fw.git
```
Comment (hayhut) [inline README.md]:
```suggestion
git clone https://github.com/SYZYGYfpga/avr-dna-fw.git
```

File: README.md [hunk 4]
```diff
@@ -106,13 +106,39 @@ configure a different parameter. For each enable "i" the bitfield contains:
...
+mkdir avr-dna-fw-build
+cd avr-dna-fw-build
+cmake ../avr-dna-fw
+make
+```
+
+To program the firmware run
```
Comment (hayhut) [inline README.md]:
```suggestion
To program the firmware run:
```

**Review by hayhut — CHANGES_REQUESTED** [review]
Two tiny suggestions, and then this looks good to merge!

File: README.md (hunk 3 above)
Comment (twam) [reply README.md]: 🙈

**Review by hayhut — APPROVED** [review]
Thanks again! This will be merged today.

**Pull request merged**